strategy/sampling/sampling_rule_manifest.go

```diff
@@ -85,20 +85,20 @@ func initSamplingRules(srm *RuleManifest) {
 
 // processManifest returns the provided manifest if valid, or an error if the provided manifest is invalid.
 func processManifest(srm *RuleManifest) error {
-	if nil == srm {
-		return errors.New("Sampling rule manifest must not be nil.")
+	if srm == nil {
+		return errors.New("sampling rule manifest must not be nil")
 	}
-	if 1 != srm.Version && 2 != srm.Version {
-		return errors.New(fmt.Sprintf("Sampling rule manifest version %d not supported.", srm.Version))
+	if srm.Version != 1 && srm.Version != 2 {
+		return fmt.Errorf("sampling rule manifest version %d not supported", srm.Version)
```

Comment:
What is the reason of using fmt.Errorf instead of errors.New in this case only ?

Reply:
because `errors.New("sampling rule manifest must not be nil")` has no format 'verbs' (e.g. `%d`)

Reply:
Makes sense!